test/e2e_node/device_plugin.go

```diff
@@ -40,156 +41,166 @@ import (
 
 const (
 	// fake resource name
-	resourceName = "fake.com/resource"
+	resourceName                 = "fake.com/resource"
+	resourceNameWithProbeSupport = "fake.com/resource2"
 )
 
 // Serial because the test restarts Kubelet
 var _ = framework.KubeDescribe("Device Plugin [Feature:DevicePlugin][NodeFeature:DevicePlugin][Serial]", func() {
 	f := framework.NewDefaultFramework("device-plugin-errors")
 
 	Context("DevicePlugin", func() {
+		By("Enabling support for Kubelet Plugins Watcher")
```

Comment:
Do you think if it is possible to wrap lines 52-205 in a testDevicePlugin(f *framework.Framework, enableProbeWatcher bool) function? Then we can get test coverage on both settings with the following test config:

var _ = framework.KubeDescribe("NVIDIA GPU Device Plugin [Feature:GPUDevicePlugin][NodeFeature:GPUDevicePlugin][Serial] [Disruptive]", func() {
        f := framework.NewDefaultFramework("device-plugin-gpus-errors")
        testDevicePlugin(f, false)
}

var _ = framework.KubeDescribe("NVIDIA GPU Device Plugin [Feature:GPUDevicePluginProbe][NodeFeature:GPUDevicePluginProbe][Serial] [Disruptive]", func() {
        f := framework.NewDefaultFramework("device-plugin-gpus-errors-probe-mode")
        testDevicePlugin(f, true)
}